src/compiler/emitter.ts

```diff
@@ -2709,9 +2725,11 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
                     }
                     else {
                         if (node.flags & NodeFlags.Default) {
+                            emitEs6ExportDefaultCompat();

```

Comment:
It was tricky to convince myself that this would not get called in ES6. To do so, I had to look at the calls to emitExportMemberAssignment. I found two:

1. emitClassLikeDeclarationBelowES6 (this one was obviously below ES6)
2. emitSignatureAndBody (this one was not obvious)

The call from emitSignatureAndBody is guarded by `!isES6ExportedDeclaration`. For isES6ExportedDeclaration to be false, any of its 3 disjuncts could be false. We want the ES6 one to be false, so let's trying to prove the other 2 true.

1. `!!(node.flags & NodeFlags.Export)`: This is true because of line 2707
2. `node.parent.kind === SyntaxKind.SourceFile`: This is true by the rules of our language. We only allow export default at a source file level, or an ambient external module. We don't emit javascript for an ambient external module, so in legal code, it must be in a source file.

I think the following would help with static reasoning. Take the above check `node.parent === currentSourceFile` and guard both the System and the default cases with that. Possibly also change it to `node.parent.kind === SyntaxKind.SourceFile` to make it even clearer.